### PR TITLE
[#626] chore(ci): skip build operator if no code changes

### PR DESCRIPTION
**Pull request by advancedxy**
### What changes were proposed in this pull request?
skip build operator if no related code change

### Why are the changes needed?
Speed up and reduce power usage for CI.
Fix: #626 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
github actions.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -24,6 +24,9 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 jobs:
+  changes:
```
Comment (kaijchen) [inline .github/workflows/build.yml:27]: Call it `build-info`?

File: .github/workflows/changes.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  workflow_call:
+    outputs:
+      kubernetes:
+        description: "whether current workflow touches deploy/kubernetes dir or not"
+        value: ${{ jobs.changes.outputs.kubernetes }}
+
+jobs:
+  changes:
+    runs-on: ubuntu-20.04
+    name: detect code change
```
Comment (kaijchen) [inline .github/workflows/changes.yml:30]: Better to find a shorter name

image

**Review by kaijchen — COMMENTED** [review]
Thanks @advancedxy for the improvement. Maybe every other workflow should depend on the `build-info` workflow. Although currently it does nothing.

For reference: https://github.com/apache/ozone/actions/runs/4203354276

**Comment by kaijchen** [comment]
I have an idea, run selected CI on PR, run full CI on master.

**Comment by kaijchen** [comment]
> Anyway, do you have more comments on this PR? I prefer the improvement to this PR should be added in later prs.

I still believe we should run full CI on master for safety and consistency, especially when some CI are skipped in PR.

Cc @jerqi @zuston @xianjingfeng WDYT?

**Comment by jerqi** [comment]
> > Anyway, do you have more comments on this PR? I prefer the improvement to this PR should be added in later prs.
> 
> I still believe we should run full CI on master for safety and consistency, especially when some CI are skipped in PR.
> 
> Cc @jerqi @zuston @xianjingfeng WDYT?

+1

**Comment by kaijchen** [comment]
> We may still need to review how CI should be ran to reduce power usage later.

Another improvement is to turn off CI on draft PRs. And trigger CI when a draft PR becomes ready to review.

**Review by kaijchen — APPROVED** [review]
LGTM, thanks @advancedxy.

**Review by jerqi — APPROVED** [review]

**Pull request merged**